### ISSUE TITLE
Slack integration

### ISSUE THREAD

**Issue by jmgasper** [topcoder-platform/topcoder-x-ui#54]
In addition to posting comments to a Github or Gitlab issue, Topcoder-X can also post to Slack when updating and creating.

### Configuration

For each project, we will add a new button to the top of the `Edit a Projects` screen to the left of `Add Labels`

The button will be called `Add Slack Integration`.  This button will authorise a Slack OAuth2 app with Topcoder-X, capturing the Slack credentials for the project.  The slack app should include a bot user that can be used to post messages to Slack.

https://api.slack.com/docs/oauth 

Once the Slack integration is complete, we will replace `Add Slack Integration` in the header with `Slack` and a checkbox, indicating the setup is complete.

### Channel

Once we have the Slack authorisation, we need to select the channel where messages will be placed.  This will be a dropdown shown on the `Edit a Project` screen that only shows up when Slack has been properly configured.

The drop down will be called `Slack Channel` and it will be populated with the list of available channels from Slack, using the `channel.list` API.

### Library

Please use the Node Slack API for this functionality: https://github.com/slackapi/node-slack-sdk 

# Messages

Messages that come from Topcoder-X should include:

* A new ticket has been created
* A ticket has been assigned
* A ticket has been marked `ready for review`
* A ticket has been marked `feedback`
* A ticket has been closed and paid

Please ensure that it will be easy to add new Slack messages in the future.

### Message format

The Slack messages should look good, using the attachments functionality in Slack messages.

Each message should include the following, where appropriate:

* Title of the git issue (as a link to Git)
* Assignee 
* Payment amount 
* Ticket status
* Topcoder challenge link (as a link to Online Review)

Here's an example of a Slack integration that has been used in the past.  Use this for reference, but make sure to include the information above:

 
<img width="776" alt="screen shot 2018-08-16 at 1 58 54 pm" src="https://user-images.githubusercontent.com/21790/44186613-bfff2500-a15d-11e8-84b7-34d2c847c01e.png">

**Comment by jmgasper** [comment]
@rootelement - Would this be helpful?  Something we should investigate further or just close it out?

**Comment by rootelement** [comment]
Leave it on the backlog. After we get updated to v5 and things are stable, we can revisit. But I definitely think there are need-to-haves that are higher priority than this nice-to-have
